ports/esp32/boards/FRI3D_BADGE_2024/mpconfigboard.h

```diff
@@ -5,7 +5,7 @@
 #define MICROPY_HW_MCU_NAME                 "ESP32S3"
 
 // Enable UART REPL for modules that have an external USB-UART and don't use native USB.
-#define MICROPY_HW_ENABLE_UART_REPL         (1)
+#define MICROPY_HW_ENABLE_UART_REPL         (0)
```

Comment:
Ik snap dat de communicator zonder deze wijziging niet kan werken maar deze gewoon uitschakelen wil zeggen dat we niet meer alle log output kunnen capturen (early boot & kernel panics), iets wat @cheops en ik wel regelmatig gebruiken. Aangezien dat doorgaans enkel bij moeilijker debugbare bugs is zouden we dat wel gewoon in lokale development builds kunnen activeren.

Ik heb wat verder onderzocht en als we `CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG=y` ook nog zetten in `sdkconfig.board` dan kunnen we `idf.py monitor` gebruiken en ook early boot en stack traces zien. Ik denk wel dat dat uitsluit dat we de badge als een USB Mass Storage kunnen gebruiken maar ik weet sowieso niet of we dat van plan waren?

@cheops wat denk jij hierover?

Reply:
_**USB Mass Storage** is a can of worms that should remain **closed** for now_

`make monitor` werkt volgens mij al?

al die usb/jtag options zijn afhankelijk van de tinyUsb configuratie in micropython
dat is dus niet meer de hardware implementatie van de chip zodra micropython is opgestart
dus volgens mij worden dan de [limitations ](https://docs.espressif.com/projects/esp-idf/en/v5.2.2/esp32s3/api-guides/usb-serial-jtag-console.html#limitations) actief en hangt het dus af van de client code (micropython tinyUSB) hoe goed het allemaal werkt.